### PR TITLE
[diffusion] Fix FLUX.2 NVFP4 scale layout on flashinfer cutlass/cudnn

### PR DESCRIPTION
**Pull request by dcampora**
ModelOptFp4LinearMethod.process_weights_after_loading only converted the NVFP4 block-scales to the TMA layout when `flashinfer_backend is None` (sgl_kernel CUTLASS fallback) or `uses_flux1_scale_layout` was set. The latter requires `not checkpoint_uses_packed_qkv`, which is False for packed-qkv checkpoints such as FLUX.2-dev-NVFP4. As a result, when the flashinfer cutlass/cudnn/auto backend was selected, FLUX.2 weight scales were left in linear layout while the kernel expects the TMA layout, producing a corrupted (constant) latent and a flat output image with no error raised.

This path is only reachable on GPUs where the default trtllm FP4 backend is unavailable (e.g. sm_120 / RTX PRO 6000), since trtllm returns earlier with its own scale shuffle. CI exercises FLUX.2 NVFP4 exclusively via trtllm on B200 (sm_100), so the cutlass/cudnn path was never covered.

Apply the TMA scale permute for any flashinfer cutlass/cudnn/auto backend in addition to the existing cases. The checkpoint stores swizzled scales; they are de-swizzled to linear earlier and the permute re-swizzles them back to TMA (the two ops are exact inverses), so FLUX.1 behavior is unchanged. Verified end-to-end: FLUX.2-dev-NVFP4 (mixed), cutlass backend, on an RTX PRO 6000 (sm_120) now produces correct images.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #28245918252](https://github.com/sgl-project/sglang/actions/runs/28245918252)
Latest PR Test (Extra): :x: [Run #28245918059](https://github.com/sgl-project/sglang/actions/runs/28245918059)

### CODE REVIEW

**Review by mickqian — APPROVED** [review]

File: python/sglang/multimodal_gen/runtime/layers/quantization/modelopt_quant.py [hunk 1]
```diff
@@ -599,8 +599,20 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
         ) and getattr(layer, "prefix", "").startswith(
             ("transformer_blocks.", "single_transformer_blocks.")
         )
-        if flashinfer_backend is None or uses_flux1_scale_layout:
-            # CUTLASS and FLUX.1 CUDNN paths need the TMA scale layout.
+        # The trtllm backend already returned above with its own scale shuffle,
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/layers/quantization/modelopt_quant.py]: could we safely remove `uses_flux1_scale_layout` now?

Comment (dcampora) [reply python/sglang/multimodal_gen/runtime/layers/quantization/modelopt_quant.py]: Yes, I have removed the if-statement.
